ExpressWash/ExpressWash/SignUpViewController.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  SignUpViewController.swift
+//  ExpressWash
+//
+//  Created by Bobby Keffury on 4/20/20.
+//  Copyright © 2020 Bobby Keffury. All rights reserved.
+//
+
+import UIKit
+
+class SignUpViewController: UIViewController {
+
+    // MARK: - Properties
+
+    private var passwordButton = UIButton(type: .custom)
+    private var confirmButton = UIButton(type: .custom)
+    private var currentTappedTextField: UITextField?
+
+    // MARK: - Outlets
+
+    @IBOutlet weak var firstNameTextField: UITextField!
+    @IBOutlet weak var lastNameTextField: UITextField!
+    @IBOutlet weak var emailAddressTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    @IBOutlet weak var minimumLabel: UILabel!
+    @IBOutlet weak var registerButton: UIButton!
+
+    // MARK: - Views
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupSubviews()
+        setupTextFields()
+
+        view.layoutIfNeeded()
+    }
+
+    // MARK: - Methods
+
+    func setupSubviews() {
+
+        let width = passwordTextField.frame.size.width
+        let rect = CGRect(x: CGFloat(width - 25), y: CGFloat(5), width: CGFloat(25), height: CGFloat(25))
+
+        if passwordTextField.isSecureTextEntry {
+            passwordButton.setTitle("●", for: .normal)
+        } else {
+            passwordButton.setTitle("○", for: .normal)
+        }
+
+        passwordButton.setTitleColor(UIColor(named: "Salmon"), for: .normal)
+        passwordButton.imageEdgeInsets = UIEdgeInsets(top: 0, left: -16, bottom: 0, right: 0)
+        passwordButton.frame = rect
+        passwordButton.addTarget(self, action: #selector(self.unhidePassword), for: .touchUpInside)
+        passwordTextField.rightView = passwordButton
+        passwordTextField.rightViewMode = .always
+    }
+
+    func setupTextFields() {
+        firstNameTextField.delegate = self
+        lastNameTextField.delegate = self
+        emailAddressTextField.delegate = self
+        passwordTextField.delegate = self
+    }
+
+    @objc func unhidePassword() {
+
+        if passwordButton.titleLabel?.text == "●" {
+            passwordTextField.isSecureTextEntry = false
+            passwordButton.titleLabel?.text = "○"
+        } else {
+            passwordTextField.isSecureTextEntry = true
+            passwordButton.titleLabel?.text = "●"
+        }
+        setupSubviews()
+    }
+
+    // MARK: - Actions
+
+    @IBAction func registerButtonTapped(_ sender: Any) {
+        guard let firstName = firstNameTextField.text, !firstName.isEmpty else { return }
```

Comment:
Maybe later (don't think it needs to be done now) we should add an alert to these or something to let the user know what they missed.